pkg/loki/modules.go

```diff
@@ -147,41 +147,40 @@ func (t *Loki) initRing() (_ services.Service, err error) {
 	return t.ring, nil
 }
 
-func (t *Loki) initGroupcache() (_ services.Service, err error) {
-	if !t.Cfg.Common.GroupCacheConfig.Enabled {
+func (t *Loki) initEmbeddedCache() (_ services.Service, err error) {
+	if !t.Cfg.QueryRange.CacheConfig.Embeddedcache.IsEnabledWithDistributed() {
 		return nil, nil
 	}
 
-	t.Cfg.Common.GroupCacheConfig.Ring.ListenPort = t.Cfg.Common.GroupCacheConfig.ListenPort
-	rm, err := cache.NewGroupcacheRingManager(t.Cfg.Common.GroupCacheConfig, util_log.Logger, prometheus.DefaultRegisterer)
+	groupConfig := cache.GroupCacheConfig{
+		Enabled:    true,
+		Ring:       t.Cfg.QueryRange.CacheConfig.Embeddedcache.Ring,
+		CapacityMB: t.Cfg.QueryRange.CacheConfig.Embeddedcache.MaxSizeMB,
+		ListenPort: t.Cfg.QueryRange.CacheConfig.Embeddedcache.ListenPort,
+	}
+
+	groupConfig.Ring.ListenPort = groupConfig.ListenPort
+
+	rm, err := cache.NewGroupcacheRingManager(groupConfig, util_log.Logger, prometheus.DefaultRegisterer)
 	if err != nil {
-		return nil, gerrors.Wrap(err, "new groupcache ring manager")
+		return nil, gerrors.Wrap(err, "new embedded-cache ring manager")
 	}
 
-	t.groupcacheRingManager = rm
-	t.Server.HTTP.Path("/groupcache/ring").Methods("GET", "POST").Handler(t.groupcacheRingManager)
+	t.embeddedcacheRingManager = rm
+	t.Server.HTTP.Path("/groupcache/ring").Methods("GET", "POST").Handler(t.embeddedcacheRingManager)
 
-	gc, err := cache.NewGroupCache(rm, t.Cfg.Common.GroupCacheConfig, util_log.Logger, prometheus.DefaultRegisterer)
+	gc, err := cache.NewGroupCache(rm, groupConfig, util_log.Logger, prometheus.DefaultRegisterer)
 	if err != nil {
 		return nil, err
 	}
 
-	t.Cfg.ChunkStoreConfig.ChunkCacheConfig.GroupCache = gc.NewGroup(
-		t.Cfg.ChunkStoreConfig.ChunkCacheConfig.Prefix+"groupcache",
-		&t.Cfg.ChunkStoreConfig.ChunkCacheConfig.GroupCacheConfig,
-		stats.ChunkCache,
-	)
-	t.Cfg.QueryRange.ResultsCacheConfig.CacheConfig.GroupCache = gc.NewGroup(
+	// We support distributed embedded cache only for results currently.
```

Comment:
I think we should clarify this, because `fifocache` can be used for results cache.
I'd actually say we should remove this comment and instead discuss this in the upgrade guide. This comment will go stale.